metron-platform/metron-data-management/src/main/java/org/apache/metron/dataloads/bulk/LeastRecentlyUsedPruner.java

```diff
@@ -1,221 +0,0 @@
-/**
- * Licensed to the Apache Software Foundation (ASF) under one
- * or more contributor license agreements.  See the NOTICE file
- * distributed with this work for additional information
- * regarding copyright ownership.  The ASF licenses this file
- * to you under the Apache License, Version 2.0 (the
- * "License"); you may not use this file except in compliance
- * with the License.  You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package org.apache.metron.dataloads.bulk;
-
-import com.google.common.base.Function;
-import com.google.common.base.Joiner;
-import org.apache.commons.cli.*;
-import org.apache.hadoop.conf.Configuration;
-import org.apache.hadoop.hbase.HBaseConfiguration;
-import org.apache.hadoop.hbase.client.Scan;
-import org.apache.hadoop.hbase.mapreduce.TableMapReduceUtil;
-import org.apache.hadoop.hbase.util.Bytes;
-import org.apache.hadoop.mapreduce.Job;
-import org.apache.hadoop.util.GenericOptionsParser;
-import org.apache.metron.dataloads.hbase.mr.PrunerMapper;
-
-import javax.annotation.Nullable;
-import java.io.IOException;
-import java.text.DateFormat;
-import java.text.SimpleDateFormat;
-import java.util.Date;
-
-public class LeastRecentlyUsedPruner {
```

Comment:
> 10. Removes the LeastRecentlyUsedPruner. This logic is exposed to the user in the script bin/threatintel_bulk_prune.sh. I had some difficulty getting the integration test working and I do not believe this is worth the effort to upgrade. I found almost no documentation around this functionality. I fully expect to initiate more discussion around this. If there is a need for this, I can work further on upgrading it.

I think deprecating an entire feature set probably requires a discussion before we accept work that completely removes it. Typically when we've done this in the past there's been at least a complementary, if not more robust, alternative to the existing functionality.

Reply:
I completely agree that deprecating a feature merits a community discussion. I'd start a discuss thread about this whole PR if there are any breaking changes.

For the record, I created the feature and would be in favor of deprecating it, but only after a discussion.

Reply:
I completely agree that deprecating a feature merits a community discussion. I'd start a discuss thread about this whole PR if there are any breaking changes.

For the record, I created the feature and would be in favor of deprecating it, but only after a discussion.

Reply:
Agreed.  I can open that discuss thread.

Reply:
See https://lists.apache.org/thread.html/6d303828e0417d370f6bcbfac1d8dca01fff64c88205c7dc5bc7bf98@%3Cuser.metron.apache.org%3E

Reply:
@cestella can you answer the discuss around what this actually does?  What the goal was?
Nice to see you ;)

Reply:
nice to see you too!  I just responded.